### ISSUE TITLE
[Question] Is it possible to exclude insight per particular method?

### ISSUE THREAD

**Issue by kniziol** [nunomaduro/phpinsights#532]
Let’s take a look at result of source code scanning:

```bash
• [Code] Unused parameter:
  src/Form/LoginType.php:19: Unused parameter $options.
```

File `src/Form/LoginType.php` mentioned by the PHP Insights contains class `LoginType` that extends `AbstractType`:

```php
final class LoginType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options): void // <——— Here is line 19
    {
        // ...
    }
}
```

Definition of the [buildForm()](https://github.com/symfony/form/blob/325f8c34c3bc58192274e64f042cfc2daad792b0/AbstractType.php#L26) method that I have to implement looks like this:

```php
abstract class AbstractType implements FormTypeInterface
{
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
    }

    // …
}
```

Whole source code of the `AbstractType` class is available here: https://github.com/symfony/form/blob/5.3/AbstractType.php.

It’s true that he `$options` parameter is unused, but is required, because of the inheritance and necessity to override the `buildForm()` method. In other words, without this parameter this compile error is raised:

`
Compile Error: Declaration of App\Form\LoginType::buildForm(Symfony\Component\Form\FormBuilderInterface $builder): void must be compatible with Symfony\Component\Form\AbstractType::buildForm(Symfony\Component\Form\FormBuilderInterface $builder, array $options)
`

How to solve the problem? Is it possible to exclude insight per particular method? 🙄

**Comment by kniziol** [comment]
I will answer myself 😏

## TL;DR

Use annotation:

```php
/**
 * @phpcsSuppress SlevomatCodingStandard.Functions.UnusedParameter
 */
```

## Details

The `\SlevomatCodingStandard\Sniffs\Functions\UnusedParameterSniff` is responsible for `[Code] Unused parameter` error and contains `private const NAME = 'SlevomatCodingStandard.Functions.UnusedParameter’;` that should be used with `@phpcsSuppress` annotation in this way:

```php
final class LoginType extends AbstractType
{
    /**
     * @phpcsSuppress SlevomatCodingStandard.Functions.UnusedParameter
     */
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        // ...
    }
}
```

**Comment by Jibbarth** [comment]
Thanks for your detailed answer. Would you want provide a PR to explain that here https://phpinsights.com/insights/code.html#unused-parameter

**Comment by kniziol** [comment]
Yes. I’ve created PR 👆
